framework/Optimizers/SPSA.py

```diff
@@ -256,29 +278,35 @@ def localEvaluateGradient(self, optVarsValues, traj,  gradient = None):
       @ Out, gradient, dict, dictionary containing gradient estimation. gradient should have the form {varName: gradEstimation}
     """
     gradArray = {}
+    # number of gradient evaluations to consider (denoising or resampling)
+    numGrads = self.gradDict['numIterForAve']
+    # prepopulate array of collected gradient
     for var in self.getOptVars(traj=traj):
-      gradArray[var] = np.zeros(0)
+      gradArray[var] = np.zeros(numGrads,dtype=object)
```

Comment:
The gradients for floats are floats, while the gradients of vectors are vectors.  This will error if you limit the `dtype` to `float` when setting vector values.  The `object` type is the only (that I know of) that accepts both.